### ISSUE TITLE
Should not need to call frame twice

### ISSUE THREAD

**Issue by aendrew** [Financial-Times/g-chartframe#62]
At the moment users call the frame twice in order to get updated positioning, which seems really unintuitive and like a serious footgun for learners.

While we're refactoring this, it would be nice to find a way of enabling users to call each chartframe once and only once. This would improve usability and also reduce code footprint.

**Comment by bobhaslett** [comment]
, I 'think' this may have something to do with the need to recall the g-axis after the labels have been measured.

**Comment by aendrew** [comment]
Yeah, definitely that, but it's still a footgun and we should try to eliminate that requirement if possible.

**Comment by bobhaslett** [comment]
Sounds good to me
